### PR TITLE
Add code coverage

### LINKED ISSUE #25: Setup code coverage
We should generate a code coverage report based on our (doc?) tests and a coverage badge to the readme and crates.io page.

### PR DESCRIPTION
**Pull request by DarkDrek**
PR to test code coverage
Will close #25

### CODE REVIEW

**Comment by DarkDrek** [comment]
Finally it's working :smiley:
Any feedback?

File: README.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 [![CircleCI](https://circleci.com/gh/milesgranger/proffer.svg?style=svg)](https://circleci.com/gh/milesgranger/proffer)
 [![crates.io](http://meritbadge.herokuapp.com/proffer)](https://crates.io/crates/proffer)
 [![License](https://img.shields.io/badge/license-Unlicense-green.svg)](http://unlicense.org/)
+[![Coverage Status](https://coveralls.io/repos/github/milesgranger/proffer/badge.svg?branch=add-code-coverage)](https://coveralls.io/github/milesgranger/proffer?branch=add-code-coverage)
```
Comment (milesgranger) [inline README.md]: Going to want to change the branch here to `master`

File: .circleci/config.yml [hunk 2]
```diff
@@ -39,8 +39,18 @@ jobs:
             rustup component add rustfmt
             cargo fmt --all -- --check
 
+  coverage:
+    machine: true
+    steps:
+        - checkout
+        - run:
+            name: Coverage with docker
+            command: docker run --security-opt seccomp=unconfined -v "${PWD}:/volume" xd009642/tarpaulin sh -c "cargo tarpaulin --ciserver circle-ci --coveralls $COVERALLS_REPO_TOKEN"
```
Comment (milesgranger) [inline .circleci/config.yml]: Just a shame that we have to do it this "ugly" way, but I see on the repo it is the suggested way, so who am I to complain. :man_shrugging: 
Thanks for adding the code coverage :+1:

File: README.md (hunk 1 above)
Comment (DarkDrek) [reply README.md]: I didn't see that :D

File: .circleci/config.yml (hunk 2 above)
Comment (DarkDrek) [reply .circleci/config.yml]: Code coverage needs special access to augment the tests for reporting.

File: README.md [hunk 3]
```diff
@@ -3,6 +3,7 @@
 [![CircleCI](https://circleci.com/gh/milesgranger/proffer.svg?style=svg)](https://circleci.com/gh/milesgranger/proffer)
 [![crates.io](http://meritbadge.herokuapp.com/proffer)](https://crates.io/crates/proffer)
 [![License](https://img.shields.io/badge/license-Unlicense-green.svg)](http://unlicense.org/)
+[![Coverage Status](https://coveralls.io/repos/github/milesgranger/proffer/badge.svg?master)](https://coveralls.io/github/milesgranger/proffer?branch=master)
```
Comment (milesgranger) [inline README.md]: Missing `branch=` on the first link

Comment (DarkDrek) [reply README.md]: Thanks again

**Pull request merged**